File: packages/api-types/src/models/party-type.ts
/**
 * 参与方类型(PartyType)域类型 —— 对齐 no.nks.dto.PartyType* 系列 DTO。
 */

/**
 * 参与方类型。
 * sortOrder / companyID 为 @JsonIgnore,不出现在线上。
 * isDefault 通过 @JsonProperty("isDefault") 显式命名。
 */
export interface PartyTypeDto {
  id?: number;
  name?: string;
  isDefault?: boolean;
  workflowCategoryID?: number;
}

/** 创建/更新请求包装。 */
export interface PartyTypeRequestDto {
  partyType?: PartyTypeDto;
}

/** 响应包装。 */
export interface PartyTypeResponseDto {
  partyType?: PartyTypeDto;
}

File: backend/src/main/java/no/nks/entity/PartyType.java
package no.nks.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "PartyType", schema = "Party")
public class PartyType {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ID")
    private Integer id;
    
    @Column(name = "Name", length = 500)
    private String name;
    
    @Column(name = "SortOrder")
    private Integer sortOrder;
    
    @Column(name = "IsDefault", nullable = false)
    private boolean isDefault;
    
    @Column(name = "WorkflowCategoryID")
    private Integer workflowCategoryId;
    
    @Column(name = "CompanyID")
    private Integer companyId;
    
    // Custom getter and setter to handle the isDefault field
    public boolean isDefault() {
        return isDefault;
    }
    
    public void setDefault(boolean isDefault) {
        this.isDefault = isDefault;
    }
} 